packages/schema/src/doc.ts

```diff
@@ -81,7 +81,32 @@ export class JupyterCadDoc
     return undefined;
   }
 
+  getDependants(name: string): string[] {
+    // Recursively get dependants of a specific object
+    // This may be quite slow with many objects, we should optimize it
+    const dependants: string[] = [];
+    for (const obj of this._objects) {
+      if ((obj.get('dependencies') || []).includes(name)) {
+        dependants.push(obj.get('name'));
+      }
+    }
+    let subDependants: string[] = [];
+    for (const dependant of dependants) {
+      subDependants = subDependants.concat(this.getDependants(dependant));
+    }
+    return dependants.concat(subDependants);
+  }
```

Comment:
Depth-first search  can help here, but i'm not sure the following code compile

```js
  getDependants(name: string): string[] {
    const dependants: string[] = [];
    const dependantMap = new Map<string, Set<string>>();

    for (const obj of this._objects) {
      const deps: string[] = obj.get("dependencies") || [];
      const objName = obj.get("name");
      deps.forEach((dep) => {
        const currentSet = dependantMap.get(dep);
        if (currentSet) {
          currentSet.add(objName);
        } else {
          dependantMap.set(dep, new Set([objName]));
        }
      });
    }
    const selectedDeps = dependantMap.get(name);
    if (!selectedDeps) {
      return [];
    }
    while (selectedDeps.size) {
      const depsList = [...selectedDeps];
      depsList.forEach((it) => {
        dependants.push(it);
        selectedDeps.delete(it);
        dependantMap.get(id)?.forEach((newIt) => selectedDeps.add(newIt));
      });
    }

    return dependants;
  }
```

Reply:
Thanks!

Reply:
does it work? I did not check the result locally